src/Google.Datastore.V1Beta3/DatastoreDbImpl.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using static Google.Datastore.V1Beta3.CommitRequest.Types;
+using static Google.Datastore.V1Beta3.ReadOptions.Types;
+
+namespace Google.Datastore.V1Beta3
+{
+    /// <summary>
+    /// Wrapper around <see cref="DatastoreClient"/> to provide simpler operations.
+    /// </summary>
+    /// <remarks>
+    /// This is the "default" implementation of <see cref="DatastoreDb"/>. Most client code
+    /// should refer to <see cref="DatastoreDb"/>, creating instances with
+    /// <see cref="BigqueryClient.Create(string, string, DatastoreClient)"/>. The constructor
+    /// of this class is public for the sake of constructor-based dependency injection.
+    /// </remarks>
+    public sealed class DatastoreDbImpl : DatastoreDb
+    {
+        /// <inheritdoc/>
+        public override DatastoreClient Client { get; }
```

Comment:
Possible nit: Should these properties really be positioned after the constructor?